clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MergeConflictResolver.java

```diff
@@ -550,7 +541,12 @@ public MergeConflictResult update(
         supersetValueSchemaEntry);
 
     int oldValueSchemaID = oldValueAndRmd.getValueSchemaID();
-    Schema oldValueSchema = getValueSchema(oldValueAndRmd.getValueSchemaID());
+    // todo(Sushant): Discuss with Jialin if we can use incomingValueSchemaId (or supersetSchemaId?) when
+    // oldValueSchemaID is -1
```

Comment:
My understanding is in semantic, when there is no old value exist, we should create a default record refilling each field with default values. Superset schema must have greater or equal fields than incoming schema so I'd suggest using superset schema here to match the semantic. Although I also understand that using incoming value schema is also fine.
But I'd actually suggest we make the change inside prepareValueAndRmdForUpdate() so that the returned oldValueAndRmd is a healthy one.